runtime/tests/integration_tests.rs

```diff
@@ -587,7 +590,7 @@ fn test_pauser_privilege_level_dpe_context_thresholds() {
         };
 
         // If we are on the last call to DeriveChild, expect that we get a PL0_ACTIVE_DPE_CONTEXT_THRESHOLD_EXCEEDED error.
-        if i == InvokeDpeCmd::PL0_DPE_ACTIVE_CONTEXT_THRESHOLD - 1 {
+        if i == num_iterations - 1 {
```

Comment:
Shouldn't this fail when `i == num_iterations` not `num_iterations - 1`?

I think this command is just succeeding because if resp is not an error, the if on line 600 never gets entered.

Would recommend adding an else there which panics

Reply:
No, because the last iteration is num_iterations - 1 (i do see how the naming could be confusing). 

But the reason this makes sense is because before any of the DeriveChild calls in this test even run, we already have 2 non-inactive contexts - context[0] is retired and created during auto initialization, and context[1] is active due to the derive child call in drivers.rs. So then, we iterate [0, 7), and iterations the six iterations 0, 1, 2, 3, 4, 5, create 6 active contexts. Then we have 8 total active contexts. Then when num_iterations - 1 = i = 6, the derive child call would fail since we would go above the threshold.

Reply:
Ok makes sense. I was confused about why the test was passing before. I still think it would be good to add an `else { panic!(...); }` after line 605.

Reply:
I did add the else clause which panics though.